src/main/java/com/thevoxelbox/voxelsniper/brush/GenerateTreeBrush.java

```diff
@@ -116,71 +106,36 @@ private void leafNodeCreate() {
                     if ((xSquared + Math.pow(y, 2) + zSquared) <= bSquared) {
                         // Chance to skip creation of a block.
-                        if (this.randGenerator.nextInt(100) >= 30) {
+                        if (this.chance(30)) {
                             // If block is Air, create a leaf block.
-                            if (this.getWorld().getBlockAt(blockPositionX + x, blockPositionY + y, blockPositionZ + z).getType() == Material.AIR) {
+                            if (location.getOffsetBlock(x, y, z).getType() == Material.AIR) {
                                 // Adds block to undo function.
-                                if (this.getBlockMaterialAt(blockPositionX + x, blockPositionY + y, blockPositionZ + z) != leavesMaterial) {
-                                    this.undo.put(this.clampY(blockPositionX + x, blockPositionY + y, blockPositionZ + z));
+                                if (location.getOffsetBlock(x, y, z).getBlockData().getMaterial() != leavesMaterial) {
+                                    this.undo.put(location.getOffsetBlock(x, y, z));
                                 }
                                 // Creates block.
-                                this.clampY(blockPositionX + x, blockPositionY + y, blockPositionZ + z).setBlockData(leavesMaterial.createBlockData(), false);
-                            }
-                        }
-                        if (this.randGenerator.nextInt(100) >= 30) {
-                            if (this.getWorld().getBlockAt(blockPositionX + x, blockPositionY + y, blockPositionZ - z).getType() == Material.AIR) {
-                                if (this.getBlockMaterialAt(blockPositionX + x, blockPositionY + y, blockPositionZ - z) != leavesMaterial) {
-                                    this.undo.put(this.clampY(blockPositionX + x, blockPositionY + y, blockPositionZ - z));
-                                }
-                                this.clampY(blockPositionX + x, blockPositionY + y, blockPositionZ - z).setBlockData(leavesMaterial.createBlockData(), false);
-                            }
-                        }
-                        if (this.randGenerator.nextInt(100) >= 30) {
-                            if (this.getWorld().getBlockAt(blockPositionX - x, blockPositionY + y, blockPositionZ + z).getType() == Material.AIR) {
-                                if (this.getBlockMaterialAt(blockPositionX - x, blockPositionY + y, blockPositionZ + z) != leavesMaterial) {
-                                    this.undo.put(this.clampY(blockPositionX - x, blockPositionY + y, blockPositionZ + z));
-                                }
-                                this.clampY(blockPositionX - x, blockPositionY + y, blockPositionZ + z).setBlockData(leavesMaterial.createBlockData(), false);
+                                location.getClampedBlock().setBlockData(leavesMaterial.createBlockData(), false);
```

Comment:
```suggestion
                                location.getOffsetBlock(x, y, z).setBlockData(leavesMaterial.createBlockData(), false);
```
Shouldn't this also be location.getOffsetBlock??

Reply:
You're right